docs/rules/remove-null-check-before-instanceof.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Remove Null Checks Before Instanceof
+description:
+    Finds and removes null checks before occurrences of instanceof. Since null is not an instance of anything, the null check is redundant.
+
+tags: ["Java 1.1", "Readability", "Coding Conventions"]
+---
+
+# Remove Null Checks Before Instanceof
+
+[[toc]]
+
+## Properties
+
+| Property                        | Value |
+|:------------------------------- |:----- |
+| First seen in jSparrow version  | [3.8.0](/eclipse/release-notes.html#_3-8-0) |
+| Minimum Java version            | 1.1   |
+| Remediation cost                | 5 min |
+| Links                           | [S4201](https://sonarcloud.io/organizations/default/rules?open=squid%3AS4201&rule_key=squid%3AS4201) |
+
+## Description
+
+Finds and removes null checks before occurrences of `instanceof`. Since `null` is not an instance of anything, the null check is redundant.
+
+## Benefits
+
+Improves readability by removing redundant code.
+
+## Requirement & Tags
+
+::: warning Requirements
+Java 1.1
+:::
+
+::: tip Tags
+<TagLinks />
+:::
+
+## Code Changes
+
+### Null-check in conjunction with instanceof
```

Comment:
In the description, null check is written without dash (-). Do we want to remove it from here too?

Reply:
Changed all occurrences of "null check" to "null-check"